File: moveElementToEnd.js
const moveElementToEnd = (array, toMove) => {
	let left = 0
	let right = array.length - 1 
	while (left < right) {
		let current = array[left]
		if (current === toMove && array[right] !== toMove) {
			array[left] = array[right]
			array[right] = current
			left++
			right--
		} else if (current === toMove && array[right] === toMove) {
			right--
		} else if (current !== toMove) {
			left++
		}
	}
  return array
}

File: Arrays/move_element_to_end.py
def move_element_to_end(array, toMove):
    left = 0
    right = len(array) - 1
    while left < right:
        while left < right and array[right] == toMove:
            right -= 1
        if array[left] == toMove:
            array[left], array[right] = array[right], array[left]
        left += 1

    return array